tutorials/cheyenne/README.md

```diff
@@ -0,0 +1,185 @@
+
+If you run into trouble with the installation, please consult the installation
+documentation [here](https://www.craylabs.org/docs/installation.html)
+
+## Examples
+
+Three of the examples utilize interactive allocations, which is the preferred method of
+launching SmartSim.
+
+When utilizing OpenMPI (as opposed to other run commands like ``srun``) to launch the
+Orchestrator database, SmartSim needs to be informed of the nodes the user would like
+the database to be launched on.
+
+This can be automated, and code for the automation of hostname aquisition is included in
+most of the files. This recipe can be followed for launching the Orchestrator with
+OpenMPI on PBS systems.
+
+
+```python
+def collect_db_hosts(num_hosts):
+    """A simple method to collect hostnames because we are using
+       openmpi. (not needed for aprun(ALPS), Slurm, etc.
+    """
+
+    hosts = []
+    if "PBS_NODEFILE" in os.environ:
+        node_file = os.environ["PBS_NODEFILE"]
+        with open(node_file, "r") as f:
+            for line in f.readlines():
+                host = line.split(".")[0]
+                hosts.append(host)
+    else:
+        raise Exception("could not parse interactive allocation nodes from PBS_NODEFILE")
+
+    # account for mpiprocs causing repeats in PBS_NODEFILE
+    hosts = list(set(hosts))
+
+    if len(hosts) >= num_hosts:
+        return hosts[:num_hosts]
+    else:
+        raise Exception(f"PBS_NODEFILE had {len(hosts)} hosts, not {num_hosts}")
+```
+
+
+### 1. launch_distributed_model.py
+
+Launch a distributed model with OpenMPI through SmartSim. This could represent
+a simulation or other workload that contains the SmartRedis clients and commuicates
+with the Orchestrator.
+
+This example runs in an interactive allocation with at least three
+nodes and 20 processors per node. be sure to include mpiprocs in your
+allocation.
+
+```bash
+qsub -l select=3:ncpus=20:mpiprocs=20 -l walltime=00:20:00 -A <account> -q premium -I
+```
+
+After obtaining the allocation, make sure to module load your conda or python environment
+with SmartSim and SmartRedis installed, as well as module load OpenMPI and gcc 8.3 as
+specified at the top of this README.
+
+Compile the simple hello world MPI program.
+
+```bash
+mpicc hello.c -o hello
+```
+
+Run the model through SmartSim in the interactive allocation
+
+```bash
+python launch_distributed_model.py
+```
+
+
+### 2. launch_database_cluster.py
+
+This file shows how to launch a distributed ``Orchestrator`` (database cluster) and
+utilize the SmartRedis Python client to communicate with it. This example is meant
+to provide an example of how users can interact with the database in an interactive
+fashion, possibly in a medium like a jupyter notebook.
+
+This example runs in an interactive allocation with at least three
+nodes and 2 processors per node. be sure to include mpiprocs in your
+allocation.
+
+```bash
+qsub -l select=3:ncpus=1 -l walltime=00:20:00 -A <account> -q premium -I
```

Comment:
Probably leave out the `-q premium` or mention that it's optional

Reply:
Actually, for all these, I would say why don't you just put this command at the top. Otherwise I can see someone just going through and blindly copy and pasting these commands for every example script.